src/helper.ts

```diff
@@ -478,3 +478,100 @@ export function splitPipPackages(installed?: ISolvedPackages) {
   }
   return { installedCondaPackages, installedPipPackages };
 }
+export function allowInstallSpecs(
+  installed: ISolvedPackages | undefined,
+  specs: string | string[] | undefined,
+  pipSpecs: string[] | undefined,
+  logger?: ILogger
+) {
+  let isAllow = true;
+  let parsedSpecs: string[] = [];
+  let parsedPipSpec: string[] = [];
+  if (typeof specs !== 'string') {
+    parsedSpecs = skipSpecs(installed, specs, logger);
+    console.log('parsedSpecs', parsedSpecs);
+    parsedPipSpec = skipSpecs(installed, pipSpecs, logger);
```

Comment:
I don't think we should skip any spec. We should just skip the solving if all new specs are already respected, but all given specs should still be respected if there is indeed a solving being done.

So I believe we can simplify this function a bit (maybe to the point that it's not a function anymore and it's simply inlined in the `solve` function).